_posts/2015-09-28-service-discovery.md

```diff
@@ -0,0 +1,57 @@
+---
+layout: post
+title: "Service discovery with Consul, Registrator & HAProxy"
+date: "2015-09-28 08:30:00"
+icon: search-plus
+categories: tech-stories
+authors: ["Dirk Jablonski"]
+---
+
+As epages is heading for a new, microservice-based architecture, new challanges arise. In this kind of architecture, there are a lot of services, which are usually deployed redundantly for high availability. But from a client perspective, all these service instances should be transparent. The client should not need to care about multiple instances of a services, their different locations on the network or if they are operational, down for maintenance or in a failure state. A client should only need to deal with a single address of the service.
```

Comment:
challanges > challenges
... multiple instances of a services > multiple instances of a service